platform/linux-generic/include/odp_llqueue.h

```diff
@@ -41,6 +41,7 @@ static odp_bool_t llq_on_queue(struct llnode *node);
  *****************************************************************************/
 
```

Comment:
Thanks for spotting this bug and submitting a fix.

At a very quick glance the fix appears valid, but I think the commit message could be made more clear and follow the ODP commit message style (separate title line, line lenghts, etc). Maybe something like this would explain the commit better:

```
avoid endless loop in llq_dequeue()

llq_enqueue() updates the next pointer of the previous tail node after it
has atomically updated the list head structure. Because of this, when
llq_dequeue() tries to dequeue from a multi-element list, it has to wait
until the next pointer of the head node is not null.

If two llq_dequeue() operations occur at the same time, it could happen
that one thread succesfully dequeues the head node when a second thread
has already read the list head structure but has not yet checked the next
pointer of the head node. Now if the first thread that already dequeued
the head node writes null to its next pointer (e.g. by enqueueing the node
to some other llqueue), the second thread may wait indefinitely for the
next pointer to become non-null. At this point the second thread is using
an outdated list head structure and any update to it through the scd
operation would fail. But the second thread does not get that far unless
the next pointer of the node that now lives outside this list gets
a non-null value.

Fix the problem by putting a loop count limit on the wait loop and
restarting the ll-sc loop if the loop count is reached.

Signed-off-by: your name <your email>
```

Reply:
Thanks for the very detailed commit message. Yes, that explained the commit better than my original message. 

I will update.

Reply:
Commit message updated.